src/main/java/gift/kakao/login/controller/KakaoLoginController.java

```diff
@@ -0,0 +1,107 @@
+package gift.kakao.login.controller;
+
+import gift.kakao.login.service.KakaoLoginService;
+import gift.user.domain.User;
+import gift.user.domain.dto.LoginRequest;
+import gift.user.domain.dto.LoginResponse;
+import gift.user.domain.dto.PointDTO;
+import gift.user.service.UserService;
+import gift.utility.JwtUtil;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.servlet.http.HttpServletRequest;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.view.RedirectView;
+
+@RestController
+@RequestMapping("/api/members")
+@Tag(name = "회원 API")
+public class KakaoLoginController {
+    @Value("${kakao.client-id}")
+    private String clientId;
+
+    @Value("${kakao.redirect-uri}")
+    private String redirectUri;
+
+    private String frontUrl;
+
+    private final UserService userService;
+    private final KakaoLoginService kakaoLoginService;
+
+    public KakaoLoginController(UserService userService, KakaoLoginService kakaoLoginService) {
+        this.userService = userService;
+        this.kakaoLoginService = kakaoLoginService;
+    }
+
+    @PostMapping("/register")
+    public ResponseEntity<LoginResponse> register(@RequestBody LoginRequest loginRequest) {
+        LoginResponse loginResponse = userService.save(loginRequest);
+
+        return new ResponseEntity<>(loginResponse, HttpStatus.CREATED);
+    }
+
+    @PostMapping("/login")
+    public ResponseEntity<LoginResponse> login(@RequestBody LoginRequest loginRequest) {
+        Optional<User> userOptional = userService.findByEmail(loginRequest.email());
```

Comment:
이렇게도 구현할 수 있을 것 같네요ㅎㅎ
```suggestion
        Boolean isLoginSuccess = userService.findByEmail(loginRequest.email())
                .filter(v -> v.isPasswordMatches(loginRequest.password()))
                .isPresent()
```